mod_dumpost.c

```diff
@@ -75,23 +87,58 @@ apr_status_t logit(ap_filter_t *f) {
       apr_ctime(time, r->request_time);
 
       // condition taken from mod_security
-#if AP_SERVER_MAJORVERSION_NUMBER > 1 && AP_SERVER_MINORVERSION_NUMBER > 2
+	  #if AP_SERVER_MAJORVERSION_NUMBER > 1 && AP_SERVER_MINORVERSION_NUMBER > 2
       char *ip = r->connection->client_ip;
-#else
+      #else
       char *ip = r->connection->remote_ip;
-#endif
-
+	  #endif
       apr_size_t nbytes_written;
-      char *text = apr_psprintf(r->pool, "[%s] %s \"%s\" %s\n",time, ip, r->the_request, state->buffer);
-      apr_status_t rc = apr_file_write_full(state->fd, text, strlen(text), &nbytes_written);
-
-      if (rc != APR_SUCCESS) {
-        ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, r, "mod_dumpost: error while writing to log");
-        return rc;
-      }
-      apr_file_close(state->fd);
+      char *text = apr_psprintf(r->pool, "[%s] %s \"%s\" ",time, ip, r->the_request);
+	  
+	  //Test buffer to search non ASCII characters
+	  int not_bin=1;
+	  for (int i = 0; i < state->log_size; i++)
+	  {
+		//if (state->buffer[i]==0) {
+		if (state->buffer[i]<0x20 || state->buffer[i]>0x7E) {
+			if (state->buffer[i]=='\n') continue;
+			if (state->buffer[i]=='\r') continue;
+			not_bin=0;
```

Comment:
So `"not_bin" == 0` mean it is binary ? this is a weird name, how about initialize it at 0 and not_bin = 1 here then ?
But honestly, I think you should drop this check -- if the user specify "I want to get hex" then we should just hex it, the data doesn't matter..